### PR TITLE
Patient Details

### LINKED ISSUE #20: ARIKE: To create an interface to capture accurate patient details during 'first visit': 1. Personal Information
There are two subsets to this, first being 'Personal information' and the second is 'Disease history'.

**Personal information**

1. State - (prefill based on user)
2. District - (prefill based on user)
3. LSG - (Panchayat/Municipality/Corporation)
4. Ward
5. Economic status: Well off/Middle class/Poor/Very Poor (dropdown)
6. Any financial support/pension/social financial support (revised list will be shared by Dr Athul) - (will confirm if it’s 1 or more)

**Patient demographics (basic details of patient)**

7. Name
8. DOB
9. Sex
10. Address
11 Phone no
12 Emergency no
13 Disease (multichoice) - Bodhi - Not sure about this. Keep as a text field for now.
14 Route map from clinic to home - exact location : (text area) - (optional)

**Family tree: (array)**

Fill the details of relatives staying with the patient
15 Name
16 Their relationship to patient - d (mother/father/son/daughter/son in law/daughter in law/grandson/granddaughter/uncle/aunt/niece/nephew)
17 Date of Birth - (optional)
18 Education - (uneducated/less than 10th standard/10th standard/12th standard/graduate/post graduate) (optional)
19 Occupation - (jobless/homemaker/public servant/private job/business) (optional)
20 Remarks - (text field) (optional)
21 What does the patient think of his/her disease and care given - (text area) - optional
22 What does the Patient’s family think of patient’s disease and care given - text area - (optional)
23 Neighbourhood doctor + nurse: (array)
     Name
     Phone
24 Volunteer from his locality (assigned volunteers near them) (array)
     Name
     Phone

### LINKED ISSUE #21: ARIKE: To create an interface to capture accurate patient details during 'first visit': 2. Disease Status
**Disease history**

1. They should be able to add multiple diseases. For each disease capture.

Disease ( select)
Table Disease = { name, ICDS code}
Seed Diseases= DM/Hypertension/IHD/COPD/Dementia/CVA/Cancer/CKD
Date of diagnosis( capture no of years + months)
Investigation: - lab reports etc (Text are)
Treatments - like chemotherapy/radiation/dialysis (Dr Athul to refer an oncologist to give the list) (select)
Status - Follow up/SOS (select)
Remarks (text area)

### PR DESCRIPTION
**Pull request by peeyush14goyal**
## Proposed Changes

Patient Information
- Add Personal Details
- Add Family Details
- Add Disease History

## Issues
closes #20  , closes #21

### CODE REVIEW

**Review by ProCode2 — COMMENTED** [review]
Partial review.
check the comments

File: app/views/patients/show.html.erb [hunk 1]
```diff
@@ -1,135 +1,87 @@
...
+    <div class="col">
+      <h3 class="text-3xl leading-6 font-medium text-gray-800 mb-8 pb-8 border-b">
+        Additional Information
+      </h3>
+
+      <div class="flex justify-between mb-8 bg-indigo-100 px-8 py-4 rounded-md">
+        <%= link_to "Personal Information", show_personal_details_path, class: "tab-label px-4 py-6 text-lg text-white bg-indigo-600 rounded hover:bg-indigo-800 w-3/12 text-center" %>
+        <%= link_to "Family Details", show_family_details_path, class: "tab-label px-4 py-6 text-lg text-white bg-indigo-600 rounded hover:bg-indigo-800 w-3/12 text-center" %>
+        <%= link_to "Disease History", show_disease_history_path, class: "tab-label px-4 py-6 text-lg text-white bg-indigo-600 rounded hover:bg-indigo-800 w-3/12 text-center" %>
+      </div>
```
Comment (ProCode2) [inline app/views/patients/show.html.erb:55]: The styles of these links are not suitable for small screens(mobile).

**Review by ProCode2 — CHANGES_REQUESTED** [review]
- [ ] Add seeds for patients details

File: app/models/patient.rb [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+    self.users = []
+    user_ids.each do |user_id|
+      self.users << User.find_by(id: user_id)
+    end
+  end
+
+  def self.get_ward(patient)
+    facility = Facility.all.find_by(id: patient.facility_id)
+    Ward.find(facility.ward)
+  end
```
Comment (ProCode2) [inline app/models/patient.rb:27]:
```suggestion
  def self.get_ward(patient)
    patient.facility.ward
  end
```

This can be written like this

**Review by ProCode2 — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
+  put "/patients/:id/patient_disease_summary/", to: "patient_disease_summaries#update"
+  # family_details
+  get "/patients/:id/family_details", to: "family_details#index", as: :family_details
+  get "patients/:id/family_details/all", to: "family_details#allMembers"
+  get "/patients/:patient_id/family_details/edit", to: "family_details#edit"
+  put "/patients/:id/family_details/", to: "family_details#update"
+  # patient information
+  get "/patients/:id/show/family_details", to: "patients#show", as: :show_family_details
+  get "patients/:id/show/personal_details", to: "patients#show", as: :show_personal_details
+  get "/patients/:id/show/disease_history", to: "patients#show", as: :show_disease_history
```
Comment (ProCode2) [inline config/routes.rb:26]: can we use namespace to group these? 
Not sure tho.
something like:
```ruby
namespace :admin do  
    resources :posts, :comments
end
```



```

**Review by ProCode2 — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 3 above)
Comment (abhinandanarya06) [reply config/routes.rb:26]: @peeyush14goyal

File: app/views/patients/show.html.erb (hunk 1 above)
Comment (peeyush14goyal) [reply app/views/patients/show.html.erb]: Made Responsive Deisgn

File: spec/requests/patient_request_spec.rb [hunk 4]
```diff
@@ -1,34 +1,63 @@
...
+  # end
+
+  # it "see single patient" do
+  #   lsg_body = LsgBody.create!(name: "Test", kind: "Municipality")
+  #   asha = User.create!(full_name: "Asha", role: User.roles[:asha], password: "01", email: "test@test.com")
+  #   patient = Patient.create!(full_name: "Mogambe khush hua", lsg_body: lsg_body.id, asha_member: asha.id, reported_by: asha.id)
+  #   get "/patients/#{patient.id}"
+  #   expect(response).to render_template("patients/show")
+  #   expect(response.body).to include(patient.full_name)
+  # end
```
[the remark this answers is not in the document]
Comment (peeyush14goyal) [reply spec/requests/patient_request_spec.rb]: Removed Comments and fixed tests

File: spec/requests/patient_request_spec.rb [hunk 5]
```diff
@@ -1,34 +1,63 @@
...
+    post "/patients", params: { patient: { full_name: "Test123", volunteer: {}, facility_id: facility.id } }
   end
+    # it "renders the list of patients" do
+  #   lsg_body = LsgBody.create!(name: "Test", kind: "Municipality")
+  #   asha = User.create!(full_name: "Asha", role: User.roles[:asha], password: "01", email: "test@test.com")
+  #   patient = Patient.create!(full_name: "Mogambe khush hua", lsg_body: lsg_body.id, asha_member: asha.id)
+  #   get "/patients"
+  #   expect(response).to render_template(:index)
+  #   expect(response.body).to include(patient.full_name)
+  # end
```
[the remark this answers is not in the document]
Comment (peeyush14goyal) [reply spec/requests/patient_request_spec.rb]: Removed Comments and fixed tests

File: spec/requests/patient_request_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (peeyush14goyal) [reply spec/requests/patient_request_spec.rb]: Removed Comments and fixed tests

**Review by arnabsen1729 — APPROVED** [review]
LGTM

**Pull request merged**